src/librustc/hir/lowering.rs

```diff
@@ -3085,6 +3099,18 @@ impl<'a> LoweringContext<'a> {
     }
 
     fn lower_struct_field(&mut self, (index, f): (usize, &StructField)) -> hir::StructField {
+        let ty = if let TyKind::Path(ref qself, ref path) = f.ty.node {
+            let t = self.lower_path_ty(
+                &f.ty,
+                qself,
+                path,
+                ParamMode::ExplicitNamed, // no `'_` in declarations (Issue #61124)
+                ImplTraitContext::disallowed()
+            );
+            P(t)
+        } else {
+            self.lower_ty(&f.ty, ImplTraitContext::disallowed())
```

Comment:
alternatively you could add a `ParamMode` argument to `lower_ty`, not sure if that is better though

Reply:
That was my first idea, but I decided against it, because it seemed inelegant to change the signature when _most_ of the `TyKind` match arm bodies don't concern themselves with `ParamMode`.